### PR TITLE
Forest & Lotus devnet

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- [x] use the new `fil-actor-states` that include the devnet CIDs https://github.com/ChainSafe/fil-actor-states/pull/124,
- [x] parameterized block delay and propagation delay so the values are not hard coded,
- [x] https://github.com/ChainSafe/forest/issues/2934 fix is merged into the branch,
- [x] https://github.com/ChainSafe/forest/pull/2935 is merged into the branch,
- [x] genesis pledge - ~~either resolved in this PR if it affects consensus or~~ create a low-priority issue -> https://github.com/ChainSafe/forest/issues/2937
- [x] refactor `compute_tipset_state_non_async` and `compute_tipset_state` so that they are one again,
- [x] network configuration no longer customisable - there is only one true devnet (defined by Lotus, oh well),
- [x] Create a compose file with the miner + lotus + forest, update instructions,

~~- [ ] Try sending some FIL to assert that Forest is usable in such a setup. It'd be great to have it in the CI in some form.~~ https://github.com/ChainSafe/forest/issues/3034

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/2816

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/state_manager/mod.rs [hunk 1]
```diff
@@ -271,15 +271,8 @@ where
...
-        .to_owned();
-
+        .to_string();
         Ok(name)
-        // let init_act = self
-        //     .get_actor(actor::init::ADDRESS, *st)?
-        //     .ok_or_else(|| Error::State("Init actor address could not be
-        // resolved".to_string()))?; let state =
-        // init::State::load(self.blockstore(), &init_act)?;
-        // Ok(state.into_network_name())
```
Comment (lemmih) [inline src/state_manager/mod.rs:282]: Let's keep this or write a note about how the chain name should ideally be read from the genesis file. (Rahul is working on this.)

**Review by lemmih — APPROVED** [review]

**Comment by elmattic** [comment]
@LesnyRumcajs if you fix merge conflicts ⚠️, I can review this PR.

File: src/networks/devnet/mod.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
```
Comment (aatifsyed) [inline src/networks/devnet/mod.rs:2]: Some of the content from the meeting earlier this week belongs here:
- what is a devnet?
- how is it expected to be used?
- design choices:
  - named devnets vs infinitely configurable ones

...etc

**Review by aatifsyed — APPROVED** [review]

File: src/networks/devnet/mod.rs (hunk 2 above)
Comment (LesnyRumcajs) [reply src/networks/devnet/mod.rs:2]: https://github.com/ChainSafe/forest/issues/3054

I'd rather this not be in the source code but in the developer handbook, but it's something secondary.

**Pull request merged**